python/dgl/data/movielens.py

```diff
@@ -0,0 +1,640 @@
+    """Check if PyTorch is the backend."""
+    if HAS_TORCH is False:
+        raise ModuleNotFoundError(
+            "MovieLensDataset requires PyTorch to be the backend."
+        )
+
+class MovieLensDataset(DGLDataset):
+    r"""MovieLens dataset for edge prediction tasks. The raw datasets are extracted from 
+    `MovieLens <https://grouplens.org/datasets/movielens/>`, introduced by
+    `Movielens unplugged: experiences with an occasionally connected recommender system <https://dl.acm.org/doi/10.1145/604045.604094>`.
+
+    The datasets consist of user ratings for movies and incorporate additional user/movie information in the form of features. 
+    The nodes represent users and movies, and the edges store ratings that users assign to movies.
+
+    Statistics:
+
+    MovieLens-100K (ml-100k)
+
+    - Users: 943
+    - Movies: 1,682
+    - Ratings: 100,000 (1, 2, 3, 4, 5)
+
+    MovieLens-1M (ml-1m)
+
+    - Users: 6,040
+    - Movies: 3,706
+    - Ratings: 1,000,209 (1, 2, 3, 4, 5)
+
+    MovieLens-10M (ml-10m)
+
+    - Users: 69,878 
+    - Movies: 10,677
+    - Ratings: 10,000,054 (0.5, 1, 1.5, ..., 4.5, 5.0)
+
+    Parameters
+    ----------
+    name: str
+        Dataset name. (:obj:`"ml-100k"`, :obj:`"ml-1m"`, :obj:`"ml-10m"`). 
+    valid_ratio: int
+        Ratio of validation samples out of the whole dataset. Should be in (0.0, 1.0).
+    test_ratio: int
+        Ratio of testing samples out of the whole dataset. Should be in (0.0, 1.0) for ml-1m and ml-10m. This parameter is invalid
+        when :obj:`name` is :obj:`"ml-100k"`, since it testing samples are pre-specified.
+        Default: None
+    raw_dir : str
+        Raw file directory to download/contain the input data directory.
+        Default: ~/.dgl/
+    force_reload : bool
+        Whether to reload the dataset. Default: False
+    verbose : bool
+        Whether to print out progress information. Default: True.
+    reverse_edge : bool
+        Whether to add reverse edges in graph. Default: True.
+    transform : callable, optional
+        A transform that takes in a :class:`~dgl.DGLGraph` object and returns
+        a transformed version. The :class:`~dgl.DGLGraph` object will be
+        transformed before every access.
+
+    Attributes
+    ----------
+    train_graph: dgl.DGLGraph
+        Training graph
+    valid_graph: dgl.DGLGraph
+        Validation graph
+    test_graph: dgl.DGLGraph
+        Testing graph
+    user_feat: torch.Tensor
+        User features
+    movie_feat: torch.Tensor
+        Movie features
+    info: dict
+        Store training, validation and testing rating pairs.
+
+    Notes
+    -----
+    - The number of edges is doubled to form an undirected(bidirected) graph structure.
+    - Each time when the dataset is loaded with a different setting of training, validation and testing ratio, the param
+    :obj:`"force_reload"` should be set to :obj:`"True"`. Otherwise the previous split of dataset will be loaded and not be
```

Comment:
```suggestion
    :obj:`force_reload` should be set to :obj:`True`. Otherwise the previous split of the dataset will be loaded and the new split will not come into effect.
```

Remove L134.